tests/frame/unpivot_test.py

```diff
@@ -20,30 +20,58 @@
     "c": [2, 4, 6],
 }
 
-expected_b_only = {
+expected_on_b_idx_a = {
     "a": ["x", "y", "z"],
     "variable": ["b", "b", "b"],
     "value": [1, 3, 5],
 }
 
-expected_b_c = {
+expected_on_b_c_idx_a = {
     "a": ["x", "y", "z", "x", "y", "z"],
     "variable": ["b", "b", "b", "c", "c", "c"],
     "value": [1, 3, 5, 2, 4, 6],
 }
 
+expected_on_none_idx_a = {
+    "a": ["x", "y", "z", "x", "y", "z"],
+    "variable": ["b", "b", "b", "c", "c", "c"],
+    "value": [1, 3, 5, 2, 4, 6],
+}
+
+expected_on_b_c_idx_none = {
+    "variable": ["b", "b", "b", "c", "c", "c"],
+    "value": [1, 3, 5, 2, 4, 6],
+}
+
+expected_on_none_idx_none = {
+    "variable": ["a", "a", "a", "b", "b", "b", "c", "c", "c"],
+    "value": ["x", "y", "z", "1", "3", "5", "2", "4", "6"],
+}
+
 
 @pytest.mark.parametrize(
-    ("on", "expected"),
-    [("b", expected_b_only), (["b", "c"], expected_b_c), (None, expected_b_c)],
+    ("on", "index", "expected"),
+    [
+        ("b", ["a"], expected_on_b_idx_a),
+        (["b", "c"], ["a"], expected_on_b_c_idx_a),
+        (None, ["a"], expected_on_none_idx_a),
+        (["b", "c"], None, expected_on_b_c_idx_none),
+        (None, None, expected_on_none_idx_none),
+    ],
 )
-def test_unpivot_on(
+def test_unpivot(
     constructor: Constructor,
     on: str | list[str] | None,
+    index: list[str] | None,
     expected: dict[str, list[float]],
+    request: pytest.FixtureRequest,
 ) -> None:
+    if on is None and index is None and "polars" not in str(constructor):
+        # TODO(): add support in other backends
+        request.applymarker(pytest.mark.xfail)
```

Comment:
Should we track this with a dedicated issue?

Reply:
True I created the issue and I didn't link it :D
https://github.com/narwhals-dev/narwhals/issues/2082

Reply:
Whops my bad! 🙈

Reply:
I'll add it in the comment before merging 👀